onnxruntime/core/framework/provider_bridge_ort.cc

```diff
@@ -1094,8 +1095,31 @@ INcclService& INcclService::GetInstance() {
 }  // namespace cuda
 #endif
 
+void UpdateProviderInfo_Tensorrt(OrtTensorRTProviderOptions* provider_options, const ProviderOptions& options) {
+  if (auto provider = s_library_tensorrt.Get()) {
+    provider->UpdateProviderOptions(reinterpret_cast<void*>(provider_options), options);
+  }
+}
+
+ProviderOptions GetProviderInfo_Tensorrt() {
+  if (auto provider = s_library_tensorrt.Get()) {
+    return provider->GetProviderOptions();
+  }
+
+  return {};
+}
+
 }  // namespace onnxruntime
 
+#ifdef USE_TENSORRT
+static char* StrDup(const std::string& str, _Inout_ OrtAllocator* allocator) {
+  char* output_string = reinterpret_cast<char*>(allocator->Alloc(allocator, str.size() + 1));
+  memcpy(output_string, str.c_str(), str.size());
+  output_string[str.size()] = '\0';
+  return output_string;
+}
+#endif
+
```

Comment:
I am not sure we want to put actual implementataion details here. It is a Bridge. Let's check with @RyanUnderhill

Reply:
If it's specific to just the bridge, it's okay, but StrDup sounds like a general function that should be somewhere more common.

Reply:
Agree. Moved StrDup to common header file.
Please note that StrDup is implemented in onnxruntime_c_api.cc as well. Any good suggestion to make only one StrDup implementation that can be used by onnxruntime_c_api.cc and provider_bridge_ort.cc? I'm thinking using the same common header file.